### ISSUE TITLE
Calling HyperSpyUI when another instace is active.

### ISSUE THREAD

**Issue by francisco-dlp** [hyperspy/hyperspyUI#130]
Currently calling hyperspyui when an instance is active results in the following:

```bash
$ hyperspyui my_file.hdf5
QLocalSocket::connectToServer: Connection refused

```

Shouldn't it open the file in the current instance instead?

**Comment by vidartf** [comment]
Yes it should, the "Connection refused" error is relevant. Which version of Qt/PyQt/PySide are you using?

**Comment by vidartf** [comment]
(the relevant file to debug is `single_application.py`)

**Comment by francisco-dlp** [comment]
qt 4.8.7, pyqt 4.11.4

**Comment by vidartf** [comment]
That should not be a problem. Try increasing `SingleApplicationWithMessaging._timeout` and see if that changes anything.

**Comment by francisco-dlp** [comment]
Still refusing with 10000

**Comment by vidartf** [comment]
Do you see any log output from the existing instance?

**Comment by francisco-dlp** [comment]
This is all I get

```
ERROR:hyperspyui.singleapplication:QLocalSocket::connectToServer: Connection refused
DEBUG:hyperspyui.singleapplication:An existing instance of HyperSpyUI is running, sending arguments to it.
```

**Comment by vidartf** [comment]
PS, this uses a [QLocalServer](http://doc.qt.io/qt-4.8/qlocalserver.html) and a [QLocalSocket](http://doc.qt.io/qt-4.8/qlocalsocket.html) to communicate from the secondary instance to the primary instance. Do you possibly have an interesting network config for localhost?

**Comment by vidartf** [comment]
> This is all I get [...]

That should be the log output from the new instance of the UI. Is there anything from the existing instance (the one that is running the full UI)?

**Comment by francisco-dlp** [comment]
No, that one doesn't print anything to the log while attempting to connect.

There is nothing special about my localhost AFAICT.

**Comment by francisco-dlp** [comment]
It works for me in Windows.

**Comment by vidartf** [comment]
Could you download [this file](https://github.com/vidartf/hyperspyUI/blob/basic-CI/tests/single_app.py) and run it from two different terminals? You will need to call it with an extra arg, e.g. `python single_app.py test`.

It should isolate the problem to a very minimal scenario.

**Comment by francisco-dlp** [comment]
I get ``Exiting early!``

**Comment by vidartf** [comment]
When running the first one?

**Comment by francisco-dlp** [comment]
No, the first one is fine ``Loaded app``

**Comment by vidartf** [comment]
For me, the outputs are:

First: 
```
python single_app.py
Loaded app
<waits here until second app gets run>
["test"]
```

Second:
```
python single_app.py test
Exiting early!
```

**Comment by francisco-dlp** [comment]
In Windows?

**Comment by vidartf** [comment]
Could you check out the [basic-CI branch](https://github.com/vidartf/hyperspyUI/tree/basic-CI) on my repo? It has a test that performs those basic steps, which passes on Travis (#131).

**Comment by vidartf** [comment]
Simply running `py.test` should do.

**Comment by francisco-dlp** [comment]
I have not been able to reproduce the issue in my laptop computer which is running Ubuntu (my office computer runs Debian testing). So, you were right, there may be an issue with my OS. I'll check once I get to the office.

**Comment by francisco-dlp** [comment]
I have boiled down the problem to running hyperspyui as an application in Gnome. I have the following desktop entry:

```desktop
[Desktop Entry]
Exec=/home/fjd29/anaconda3/bin/hyperspyui %F
Name=HyperSpy UI
Terminal=false
MimeType=application/x-hdf;application/x-dm3;
Icon=/home/fjd29/Python/hyperspyUI/hyperspyui/images/hyperspy.svg
Type=Application
X-MultipleArgs=true
Categories=Science;Physics;DataVisualization;
```

I only experience the issue when I start HyperSpyUI using this method. If I change ``Terminal`` to ``true`` I don't experience the issue any more (but I  get a terminal with the app, which I don't want). When the issue triggers I have to restart in order to be able to launch hyperspyui from a terminal accepting connections. Killing hyperspyui manually doesn't work, nor login out from the gnome session.

Note: after I manage to integrate hyperspyui successfully in my gnome desktop I'll make a PR.

**Comment by vidartf** [comment]
@francisco-dlp : Could you try to print `sys.argv` on launch? I think the code might be making some assumptions about the form of that!

**Comment by vidartf** [comment]
No wait, that wouldn't explain the network error :-/
